domain-server/src/DomainServerExporter.cpp

```diff
@@ -0,0 +1,456 @@
+            }
+
+
+            continue;
+        }
+
+        if ( BLACKLIST.contains(metric_name)) {
+            continue;
+        }
+
+        bool conversion_ok = false;
+        double converted = 0;
+
+        if ( val.isString() ) {
+            // Prometheus only deals with numeric values. See if this string contains a valid one
+
+            QString tmp = val.toString();
+            converted = tmp.toDouble(&conversion_ok);
+
+            if ( !conversion_ok ) {
+                qCWarning(domain_server_exporter) << "Failed to convert value of " << orig_metric_name << " (" << metric_name  << ") to double: " << tmp << "'";
+                continue;
+            }
+
+        }
+
+        stream << QString("\n# HELP %1 %2 -> %3\n").arg(metric_name).arg(original_path).arg(iter.key());
+
+        if ( TYPE_MAP.contains(metric_name )) {
+            stream << "# TYPE " << metric_name << " ";
+            switch( TYPE_MAP[metric_name ]) {
+            case DomainServerExporter::MetricType::Untyped:
+                stream << "untyped"; break;
+            case DomainServerExporter::MetricType::Counter:
+                stream << "counter"; break;
+            case DomainServerExporter::MetricType::Gauge:
+                stream << "gauge"; break;
+            case DomainServerExporter::MetricType::Histogram:
+                stream << "histogram"; break;
+            case DomainServerExporter::MetricType::Summary:
+                stream << "summary"; break;
+            }
+            stream << "\n";
+        } else {
+            qCWarning(domain_server_exporter) << "Type for metric " << orig_metric_name << " (" << metric_name << ") not known.";
+        }
+
+        stream << path << "_" << key;
+        if (!labels.isEmpty() ) {
+            stream << "{";
+
+            bool is_first = true;
+            QHashIterator<QString,QString> iter(labels);
+
+            while( iter.hasNext() ) {
+                iter.next();
+
+                if ( ! is_first ) {
+                    stream << ",";
+                }
+
+                QString value = iter.value();
+                value.replace("\\", "\\\\");
+                value.replace("\"", "\\\"");
+                value.replace("\n", "\\\n");
+
+                stream << iter.key() << "=\"" << value << "\"";
+
+                is_first = false;
+            }
+            stream << "}";
+        }
+
+        stream << " ";
+
+        if ( val.isBool() ) {
+            stream << ( iter.value().toBool() ? "1" : "0" );
+        } else if ( val.isDouble() ) {
+            stream << val.toDouble();
+        } else if ( val.isString() ) {
+            // Converted above
+            stream << converted;
+        } else {
+            qCWarning(domain_server_exporter) << "Can't convert metric " << orig_metric_name << "(" << metric_name << ") with value " << val;
+        }
+
+        stream << "\n";
+    }
+}
+
```

Comment:
spacing in this file does not match the general codebase style. please have a look through the style guide